emails.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
+from logging import Logger
+
+from azure.communication.email import EmailClient
+
+
+class EmailsException(Exception):
+    """Base class for all email-related exceptions."""
+
+
+class EmailSendingFailedError(EmailsException):
+    """Custom Exception for email sending errors."""
+
+
+class PollingTimeoutError(EmailsException):
+    """Custom Exception for polling time-out during email sending."""
+
+
+class UnexpectedEmailSendError(EmailsException):
+    """Custom Exception for unexpected errors."""
+
+
+@dataclass
+class EmailConfig:
+    email_client: EmailClient
+    sender_email: str
+    subject: str
+    validation_pattern: str
+    email_send_success: str
+    html_content: str = "<html><h1>{}</h1></html>"
+    poller_wait_time: int = 10
```

Comment:
default parameters are fine because they make the function simpler to use and reuse. However, these should be module-level constants that are also used as the default value when lookup up the matching environment variable.